### PR TITLE
DMEERX-587,-588 RAD F2F initial commit

### PR DESCRIPTION
**Pull request by cwanmtr**
Notes:
1. A number of sub-questionnaires, such as PatientInfo, PractionerInfo, ReviewOfSystem, etc. were used in the questionnaire. However, the PhysicalExam sub-questionnaire was not used because it does not include some specific required items that are present in the RAD F2F template.

2. Test data for this RAD F2F Encounter ruleset has been added to the test-ehr project. You can use Patient Theodor Roosevelt (pat014) and select E0470 DeviceRequest for testing this ruleset.

UPDATE: I have changed to use the PhysicalExam and VitalSigns sub-questionnaires and also the EncounterWithReevaluation sub-questionnaire which was created for the Papvs #49 PR. Hence, this PR depends on the #49 PR.

### CODE REVIEW

**Comment by yuliumitre** [comment]
@cwanmtr Do you have a PR for test-ehr repo for the test data?

**Comment by yuliumitre** [comment]
For physical exam, can we use the subquestionnires of vitalsign-physicalexam, then add the additional items e.g. neck circumference and body mass?

**Comment by cwanmtr** [comment]
> @cwanmtr Do you have a PR for test-ehr repo for the test data?

I added the test data to the test-ehr master branch. I should have created a PR. I did it too fast and didn't think about the need of a PR.

File: RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesFaceToFacePrepopulation-0.1.0.cql [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+parameter device_request DeviceRequest
+
+
+context Patient
+
+define Today: Today()
+
+
+define "EncounterDate":
```
Comment (yuliumitre) [inline RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesFaceToFacePrepopulation-0.1.0.cql]: For EncounterDate, it should be the Encounter referred by the DeviceRequest if it exists.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json [hunk 2]
```diff
@@ -0,0 +1,1475 @@
...
+                  "question": "6.1",
+                  "operator": "=",
+                  "answerBoolean": true
+                }
+              ]
+            },
+            {
+              "linkId": "6.1.2",
+              "text": "Forced vital capacity in % predicted:",
+              "type": "quantity",
```
Comment (yuliumitre) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json]: We need to be consistent when it comes to display the lab results, etc.  Question for @yunwwang 
Do we want to add extension for this kind of display to limit the value as integer or decimal only? The UI provides some validation.  Or we should use the type "quantity"? It only displays the value portion of the Quantity if we use "quantity" as type. 
 {
                      "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-unit",
                      "valueCoding": {
                        "display": "%"
                      }
                    }

File: RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesFaceToFacePrepopulation-0.1.0.cql (hunk 1 above)
Comment (cwanmtr) [reply RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesFaceToFacePrepopulation-0.1.0.cql]: I was just following the comment "Find most recent encounter" given under the 'Notes' column in the 'DRLS Rule Sets - Primary DRAFT.xlsx' file. 
I will make the change like this: First find the Encounter referred by the DeviceRequest. If DeviceRequest does not point to any Encounter, find the most recent encounter. If no encounters exist, use Today.

**Comment by cwanmtr** [comment]
> For physical exam, can we use the subquestionnires of vitalsign-physicalexam, then add the additional items e.g. neck circumference and body mass?

I have not seen an example showing that. My understanding is that we cannot override any part or add a new part to a subquestionnaire. Are you suggesting that I should add some additional items right after the whole subquestionnaire like what is showed below. My concern for doing that way is that the items are out of order.
    {
      "linkId": "13",
      "type": "group",
      "text": "Physical examination",
      "item": [
    {
      "linkId": "13.1",
      "text": "Physical examination",
      "extension": [
        {
          "url": "http://hl7.org/fhir/StructureDefinition/sub-questionnaire",
          "valueCanonical": "questionnaire/physical-exam"
        }
      ],
      "type": "display",
      "required": false
    },
            {
              "required": true,
              "linkId": "13.2",
              "text": "Neck circumference (cm)",
              "extension": [
                {
                  "url": "http://hl7.org/fhir/StructureDefinition/cqf-expression",
                  "valueExpression": {
                    "language": "text/cql",
                    "expression": "\"RespiratoryAssistDevicesFaceToFacePrepopulation\".NeckCircumf"
                  }
                }
              ],
              "type": "string"
            },
            {
              "required": true,
              "linkId": "13.3",
              "text": "Body mass index (kg/m2)",
              "extension": [
                {
                  "url": "http://hl7.org/fhir/StructureDefinition/cqf-expression",
                  "valueExpression": {
                    "language": "text/cql",
                    "expression": "\"RespiratoryAssistDevicesFaceToFacePrepopulation\".BMI"
                  }
                }
              ],
              "type": "string"
            }
      ]
    },

**Comment by yuliumitre** [comment]
Yeah, adding the extra items at the end of the sub-questionnaire is sort of out of order. Maybe add the extra items before the subquestionnaire physical exam, then group them together? We can check with rest of the team as well.

**Comment by yuliumitre** [comment]
I am seeing the same/similar section from CPAP FaceToFace for Encounter date and verify if the encounter is for the current order; then if yes is answered,  the "reevaluation" is embedded.  I will create a sub-questionnaire for it  in my story.

**Comment by yuliumitre** [comment]
The changes look fine. One more thing, for lab results and quantity displays, we should not include the unit in the text box. This is not consistent with other quantity boxes. 

Screen Shot 2020-07-29 at 9 40 11 AM

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json [hunk 3]
```diff
@@ -0,0 +1,901 @@
...
+      "url": "http://hl7.org/fhir/StructureDefinition/cqf-library",
+      "valueCanonical": "http://hl7.org/fhir/us/davinci-dtr/Library/BasicClinicalInfo-prepopulation"
+    },
+    {
+      "url": "http://hl7.org/fhir/StructureDefinition/cqf-library",
+      "valueCanonical": "http://hl7.org/fhir/us/davinci-dtr/Library/BasicPatientInfo-prepopulation"
+    },
+    {
+      "url": "http://hl7.org/fhir/StructureDefinition/cqf-library",
+      "valueCanonical": "http://hl7.org/fhir/us/davinci-dtr/Library/BasicPractitionerInfo-prepopulation"
```
Comment (yuliumitre) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json]: We should not need to include BasicPatient and BasicPractitioner library anymore.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json [hunk 4]
```diff
@@ -0,0 +1,901 @@
...
+                "expression": "\"RespiratoryAssistDevicesFaceToFacePrepopulation\".RADCodings"
+              }
+            }
+          ],
+          "repeats": true,
+          "answerOption": []
+        },
+        {
+          "linkId": "4.2",
+          "text": "Other Diagnoses:",
```
Comment (yuliumitre) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json:82]: We need a text box to "describe" the other diagnoses in this group.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json (hunk 2 above)
Comment (yunwwang) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json]: Don't use quantity

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json [hunk 5]
```diff
@@ -0,0 +1,901 @@
...
+        }
+      ],
+      "type": "display"
+    },
+    {
+      "linkId": "3",
+      "extension": [
+        {
+          "url": "http://hl7.org/fhir/StructureDefinition/sub-questionnaire",
+          "valueCanonical": "questionnaire/encounter-with-reevaluation"
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json:52]: Is this a new sub questionnaire?

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json [hunk 6]
```diff
@@ -0,0 +1,901 @@
...
+              }
+            }
+          ],
+          "type": "date"
+        },
+        {
+          "linkId": "5.5",
+          "text": "Was study performed in a facility?",
+          "type": "choice",
+          "answerOption": [
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json]: Use YesNo valueset

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json (hunk 5 above)
Comment (cwanmtr) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json:52]: The encounter-with-reevaluation sub-questionnaire was created by Yolanda. See PR #49.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json (hunk 2 above)
Comment (cwanmtr) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json]: In the case of displaying the Neck Circumference value, I was not able to load the questionnaire successfully when I used type:decimal. I tried different ways and used same code from other ruleset but never got success. Not wanting to spend time to debug the problem, I just stayed with using type:quantity which works fine for my ruleset.

Comment (yuliumitre) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json]: We are not using quantity in table and it seems quantity works well here. Also, we don't really need to pre-populate physical exam section.

Comment (yuliumitre) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesFaceToFace.json]: @cwanmtr I think another benefits to use the extension of questinnaire-unit is that the unit is automatically shown on the UI. Please see the screen shot below and it also performs the client side validation for number. With "quantity", we don't have the benefit. 

Screen Shot 2020-07-31 at 8 53 05 AM


In addition, we don't need to pre-populate the fields of physical exam etc.

**Comment by cwanmtr** [comment]
Thank you Yolanda and Yunwei for your comments. Let me know if you have further comments. If not, then this PR is ready to be merged into master.

**Comment by lpdecelles** [comment]
Hello Caleb, this PR has conflicts. It might just need a forward merge from mater.

File: RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesRule-0.1.0.cql [hunk 7]
```diff
@@ -26,8 +26,13 @@ define RESULT_InfoLink:
 define RESULT_QuestionnaireOrderUri:
     'Questionnaire/RespiratoryAssistDevices'
 
+<<<<<<< HEAD
+define RESULT_QuestionnaireFaceToFaceUri:
+  'Questionnaire/RespiratoryAssistDevicesFaceToFace'
+=======
 define RESULT_QuestionnaireLabUri:
     'Questionnaire/RespiratoryAssistDevicesLab'
+>>>>>>> master
```
Comment (yuliumitre) [inline RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesRule-0.1.0.cql:34]: Please resolve the conflicts.

**Review by yuliumitre — APPROVED** [review]

**Pull request merged**